src/Amadevus.RecordGenerator.Generators/SyntaxExtensions.cs

```diff
@@ -148,6 +148,13 @@ public static NameSyntax GetTypeSyntax(this TypeDeclarationSyntax typeDeclaratio
             return SyntaxFactory.GenericName(identifier, typeArgList);
         }
 
+        public static string GetQualifiedName(this ISymbol symbol)
+        {
+            var symbolDisplayFormat = new SymbolDisplayFormat(
+                typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces);
+            return symbol.ToDisplayString(symbolDisplayFormat);
```

Comment:
This needs to be tested with nullables in both full-format and sugar-format (`Nullable<int>` and `int?`).

Reply:
This I don't expect it to make a difference, because we're working agains semantic data here (sugar shouldn't matter)
But I will surely test it 👍

Reply:
This I don't expect it to make a difference, because we're working agains semantic data here (sugar shouldn't matter)
But I will surely test it 👍

Reply:
The reason I ask is because I don't really know the behavior of those options. Since the `generic` option by default says "none", will `int?` be printed as `System.Nullable`, `System.Nullable<int>`, `int?` (don't think so), or ``System.Nullable`1``?

Reply:
There are also some readily available formats on the `SymbolDisplayFormat` type, maybe one of those would be enough? Otherwise, this is something worth making a static field for.

Reply:
Yes there are some more readable methods, but actually no they do not return the full qualified name for internal types.
A static field sounds like a good idea.

I will look into this later, but as far as I've seen it always return the name with a ? as suffix. The symbol should no be coupled to the syntax